plugins/optimization-detective/optimization.php

```diff
@@ -171,38 +169,45 @@ function od_optimize_template_output_buffer( string $buffer ): string {
 	 *
 	 * @param OD_Tag_Visitor_Registry         $tag_visitor_registry Tag visitor registry.
 	 * @param OD_URL_Metrics_Group_Collection $group_collection     URL Metrics Group collection.
-	 * @param OD_Preload_Link_Collection      $preload_links        Preload links collection.
+	 * @param OD_Link_Collection              $link_collection      Link collection.
 	 */
-	do_action( 'od_register_tag_visitors', $tag_visitor_registry, $group_collection, $preload_links );
+	do_action( 'od_register_tag_visitors', $tag_visitor_registry, $group_collection, $link_collection );
+
+	$current_tag_bookmark = 'optimization_detective_current_tag';
 
-	$visitors  = iterator_to_array( $tag_visitor_registry );
-	$generator = $walker->open_tags();
-	while ( $generator->valid() ) {
+	$visitors = iterator_to_array( $tag_visitor_registry );
+	while ( $processor->next_open_tag() ) {
```

Comment:
Initially I had this as `$processor->next_tag()` which was not correct as it was resulting in visits to closing tags. Then I added a `$processor->is_tag_closer()` check which would `continue` if `true`. But then I remembered this subclass's `next_open_tag()` helper method which "works a treat".